src/clj/rems/application/eraser.clj

```diff
@@ -1,65 +1,43 @@
 
-(defn- is-expired?
-  [application]
-  (let [last-activity (:application/last-activity application)
-        state (:application/state application)
-        expiration-threshold (some-> env :application-expiration state Period/parse)]
-    (if expiration-threshold
-      (time/before? last-activity (time/minus (time/now) expiration-threshold))
-      false)))
-
-(deftest test-is-expired
-  (with-redefs [env {:application-expiration {:application.state/draft "P90D"
-                                              :application.state/closed "P7D"}}]
-    (testing "should identify expired application by state"
-      (let [over-90d-ago (time/minus (time/now) (time/days 90) (time/seconds 1))
-            over-7d-ago (time/minus (time/now) (time/days 7) (time/seconds 1))
-            over-1d-ago (time/minus (time/now) (time/days 1) (time/seconds 1))]
-        (is (is-expired? {:application/state :application.state/draft
-                          :application/last-activity over-90d-ago}))
-        (is (not (is-expired? {:application/state :application.state/draft
-                               :application/last-activity over-7d-ago})))
-        (is (is-expired? {:application/state :application.state/closed
-                          :application/last-activity over-7d-ago}))
-        (is (not (is-expired? {:application/state :application.state/closed
-                               :application/last-activity over-1d-ago})))
-        (is (not (is-expired? {:application/state :application.state/rejected
-                               :application/last-activity over-90d-ago})))))))
-
-(defn remove-expired-applications!
-  []
-  (log/info :start #'remove-expired-applications!)
-  (doseq [app-id (->> (applications/get-all-unrestricted-applications)
-                      (filter is-expired?)
-                      (map :application/id))]
-    (try
-      (applications/delete-application! app-id)
-      (log/info "Succesfully removed application" app-id)
-      (catch Throwable t
-        (log/error "Failed to remove application" app-id ":" t))))
+(defn process-applications! []
+  (log/info :start #'process-applications!)
+  ; check that bot user exists, else log missing
+  (if (users/user-exists? expirer-bot/bot-userid)
+    (doseq [application (applications/get-all-unrestricted-applications)]
+      (when-some [cmd (expirer-bot/run-expirer-bot application)]
+        (command/command! cmd)))
+    (log/warn (str "Cannot process applications, because user " expirer-bot/bot-userid " does not exist")
+              #'process-applications!))
   (applications/reload-cache!)
-  (log/info :finish #'remove-expired-applications!))
+  (log/info :finish #'process-applications!))
 
 (mount/defstate expired-application-poller
   :start (when (:application-expiration env)
-           (scheduler/start! "expired-application-poller" remove-expired-applications! (.toStandardDuration (time/days 1))))
+           (scheduler/start! "expired-application-poller"
```

Comment:
Should we rename to `eraser-poller` to be consistent with the ns? Or is this perhaps still clearer?